### PR TITLE
feat : added dynamic selection of SQ pins in PWM mode.

### LINKED ISSUE #1100: the dynamic selection of the SQ pins for PWM pins is not there
**Actual Behaviour**

Currently in the PWM mode all the SQ pins are activated for testing purposes.

**Expected Behaviour**

The user should be able to activate or deactivate the SQ pins when producing the PWM signal

**Steps to reproduce it**

WaveGenerator -> Implement the functionality of Dynamic activation of SQ pins while producing the PWM signal

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Avjeet**
Fixes #1100 

**Changes**:
 - When used in PWM mode the SQ pins for which the value has been set will automatically be activated  
    for output

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[app-debug.zip](https://github.com/fossasia/pslab-android/files/2129936/app-debug.zip)

### CODE REVIEW

**Comment by Avjeet** [comment]
@CloudyPadmal @Vikum94 @harsh-2711 @abhinavraj23 plz only review the second commit , first one is from previous PR.

File: app/src/main/res/values/colors.xml [hunk 1]
```diff
@@ -13,6 +13,7 @@
     <color name="back_circle_color">#EDEDED</color>
     <color name="indicator_color">#d32f2f</color>
     <color name="progress_secondary_color">#EEEEEE</color>
+    <color name="purple_btn">#6002ee</color>
```
Comment (CloudyPadmal) [inline app/src/main/res/values/colors.xml]: @Avjeet can purple cross match with red? Besides on a black background it's hard to see a purple letters and what if we use the same red as in our theme?

**Comment by Avjeet** [comment]
@CloudyPadmal well red will also be not visible on black background so I used Green color.

**Comment by Avjeet** [comment]
@CloudyPadmal well, the orange is when u press the button to change its properties like duty or frequency.
The blue color will only be used when you are in PWM mode and you long press the SQ pin, that will activate that pin, the blue coloring of the SQ pins on monitor is analogous to the SQ checkboxes in initial layout.

**Comment by CloudyPadmal** [comment]
@Avjeet I don't know, doesn't it sound too complicated? So many settings and color changes for a simple wave generator. Can't we have a simple implementation? Do you think a school kid will be able to go through all that complicated setting up process? Long pressings; mode switching... @harsh-2711 @abhinavraj23 @Vikum94 please review this!

**Comment by Vikum94** [comment]
I agree with @CloudyPadmal I also didn't get why a long press is needed. Can't we use the set button for that also @Avjeet

**Comment by Avjeet** [comment]
@CloudyPadmal Ok I will remove the long press and the channels for which the value is set will be activated and for those, no value has been set will remain deactivated.

**Pull request merged**